chain/solana/src/runtime/abi.rs

```diff
@@ -0,0 +1,439 @@
+            })
+            .collect();
+        let content = content?;
+        Ok(AscStringArray(Array::new(&*content, heap)?))
+    }
+}
+
+#[test]
+fn block_to_asc_ptr() {
+    use graph::data::subgraph::API_VERSION_0_0_5;
+    use graph::prelude::hex;
+
+    let mut heap = BytesHeap::new(API_VERSION_0_0_5);
+    let block = codec::Block {
+        id: hex::decode("01").expect(format!("Invalid hash value").as_ref()),
+        number: 0,
+        version: 0,
+        previous_id: vec![],
+        previous_block: 0,
+        genesis_unix_timestamp: 0,
+        clock_unix_timestamp: 0,
+        root_num: 0,
+        last_entry_hash: vec![],
+        transactions: vec![],
+        transaction_count: 0,
+        has_split_account_changes: false,
+        account_changes_file_ref: "".to_string(),
+    };
+
+    let asc_block = asc_new(&mut heap, &block);
+    assert!(asc_block.is_ok());
+}
+
+#[test]
+fn transaction_to_asc_ptr() {
+    use graph::data::subgraph::API_VERSION_0_0_5;
+    use graph::prelude::hex;
+
+    let mut heap = BytesHeap::new(API_VERSION_0_0_5);
+    let transaction = codec::Transaction {
+        id: vec![],
+        index: 0,
+        additional_signatures: vec![],
+        header: None,
+        account_keys: vec![],
+        recent_blockhash: vec![],
+        log_messages: vec![],
+        instructions: vec![],
+        failed: false,
+        error: None,
+    };
+
+    let asc_transaction = asc_new(&mut heap, &transaction);
+    assert!(asc_transaction.is_ok());
+}
+
+#[test]
+fn instruction_to_asc_ptr() {
+    use graph::data::subgraph::API_VERSION_0_0_5;
+    use graph::prelude::hex;
+    use prost::Message;
+
+    let balance_change_1 = codec::BalanceChange {
+        pubkey: hex::decode("01").expect(format!("Invalid hash value").as_ref()),
+        prev_lamports: 1,
+        new_lamports: 2,
+    };
+
+    let account_change_1 = codec::AccountChange {
+        pubkey: hex::decode("01").expect(format!("Invalid hash value").as_ref()),
+        prev_data: hex::decode("02").expect(format!("Invalid hash value").as_ref()),
+        new_data: hex::decode("03").expect(format!("Invalid hash value").as_ref()),
+        new_data_length: 1,
+    };
+
+    let mut heap = BytesHeap::new(API_VERSION_0_0_5);
+    let instruction = codec::Instruction {
+        program_id: hex::decode("01").expect(format!("Invalid hash value").as_ref()),
+        account_keys: vec![
+            hex::decode("01").unwrap().encode_to_vec(),
+            hex::decode("02").unwrap().encode_to_vec(),
+        ],
+        data: hex::decode("01").expect(format!("Invalid hash value").as_ref()),
+        ordinal: 1,
+        parent_ordinal: 2,
+        depth: 42,
+        balance_changes: vec![balance_change_1],
+        account_changes: vec![account_change_1],
+        failed: false,
+        error: None,
+    };
+
+    let asc_instruction = asc_new(&mut heap, &instruction);
+    assert!(asc_instruction.is_ok());
+}
+
+struct BytesHeap {
```

Comment:
This is shared with NEAR now, we should move it to a common shared place so it can be re-used by both tests.